rayon-core/src/join/mod.rs

```diff
@@ -120,13 +120,12 @@ where
             worker: worker_thread.index()
         });
 
+        let latch = SpinLatch::new();
+
         // Create virtual wrapper for task b; this all has to be
         // done here so that the stack frame can keep it all live
         // long enough.
-        let job_b = StackJob::new(
-            |migrated| oper_b(FnContext::new(migrated)),
-            SpinLatch::new(),
-        );
+        let job_b = StackJob::new(|migrated| oper_b(FnContext::new(migrated)), &latch);
```

Comment:
Why did you make this one a reference? A `SpinLatch` is just an `AtomicBool`, so it seems all you're doing is adding needless pointer indirection in this `StackJob`.